Codeinwp/neve
# Feat/custom global colors

### Summary

This PR adds a custom global color adding ability. That global custom colors can be used in Gutenberg, Elementor, Beaver Plugin


**Helper Review Notes for @abaicus :**
- Data storage of global custom colors in a new theme mod: neve_global_custom_colors
    - The sample data: https://vertis.d.pr/i/9Rnuqz
    - Sanitization method: https://github.com/Codeinwp/neve/blob/bd26887f0ca01d78d712fbc8aeba95a9f5f57bc3/inc/customizer/options/colors_background.php#L158-L187
- For the label change feature: I preserved the color slug to avoid over-engineering (to remove needs of safeguards such as duplicate checks etc, also leverage by auto slug assign)
- For resolving Gutenberg CSS classes such as .has-…..-color; (the dynamic version of the `assets/scss/main/_global_colors.scss`) I leveraged `neve_dynamic_style_output` WP filter since, as I know, there is no luck to leverage Dynamic Style Generator since it depends to theme mod value for each prop and there is no callback support to customize it I think. Is that good?
- Currently, if a custom color is removed, the default color is shown as a fallback natively. So, I didn’t add any support there. Is that good?
- For global custom colors, in the beginning, I used selective refresh support, but “global colors” works with an entire refresh; therefore, a conflict occurs. If I change a custom color on the color picker, it’s applied immediately, then play with global colors; during the entire refresh, previous custom colors are shown until the customizer DOM is entirely loaded. Therefore I decided to use the entire refresh for global custom colors too. More details: https://vertis.d.pr/v/3jd1Iv


### Will affect visual aspect of the product

YES

### Screenshots 
<img width="1284" alt="Screen Shot 2023-01-02 at 19 31 40" src="https://user-images.githubusercontent.com/25361626/210280446-a4f26302-fe8d-42c0-b273-907784a8076c.png">
<img width="1388" alt="Screen Shot 2023-01-02 at 19 31 46" src="https://user-images.githubusercontent.com/25361626/210280420-f1361f4c-d486-41b5-8543-a10f88aa554a.png">
<img width="1274" alt="Screen Shot 2023-01-02 at 19 32 03" src="https://user-images.githubusercontent.com/25361626/210280424-716bfb12-97d2-4aca-873d-125063e20765.png">

![Screen Shot 2023-01-07 at 15 52 43](https://user-images.githubusercontent.com/25361626/211151582-accd608e-3abb-4dca-a11f-0417105af45a.png)


### Test instructions

- Install only Neve (the feature is completely free but please also test with PRO)
- Visit customizer->Colors&Background, you'll see the Custom Colors accordion.
- Adding&deleting custom color abilities should work well.
- Added custom colors are palette-independent. (Palette change does not affect custom colors)
- Global colors should be available in three editors(Gutenberg, Elementor, Beaver) like the global colors we currently have.
- Max custom colors limit is 30; more than 30 should be denied.


Closes https://github.com/Codeinwp/neve-pro-addon/issues/2117.
